src/payment/strategies/digitalriver/digitalriver-payment-strategy.ts

```diff
@@ -0,0 +1,167 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { Customer } from '../../../customer';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import DigitalRiverJS, { DigitalRiverDropIn, DigitalRiverInitializeToken, OnCancelOrErrorResponse, OnReadyResponse, OnSuccessResponse } from './digitalriver';
+import DigitalRiverPaymentInitializeOptions from './digitalriver-payment-initialize-options';
+import DigitalRiverScriptLoader from './digitalriver-script-loader';
+
+export default class DigitalRiverPaymentStrategy implements PaymentStrategy {
+    private _digitalRiverJS?: DigitalRiverJS;
+    private _digitalRiverDropComponent?: DigitalRiverDropIn;
+    private _submitFormEvent?: () => void;
+    private _loadSuccessResponse?: OnSuccessResponse;
+    private _digitalRiverCheckoutId?: string;
+    private _digitalRiverInitializeOptions?: DigitalRiverPaymentInitializeOptions;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _digitalRiverScriptLoader: DigitalRiverScriptLoader
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        this._digitalRiverInitializeOptions = options.digitalriver;
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(options.methodId));
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+        let clientToken: DigitalRiverInitializeToken;
+
+        if (!paymentMethod.clientToken) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        try {
+            clientToken = JSON.parse(paymentMethod.clientToken);
+        } catch (error) {
```

Comment:
is this catch worth it? you are just wrapping the thrown error into a generic error, which doesn't seem to be adding much value.